src/Adapter/LegacyContextLoader.php

```diff
@@ -0,0 +1,165 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter;
+
+use Context;
+use Currency;
+use DummyAdminController;
+use Employee;
+use Shop;
+
+/**
+ * Helps loading specific context, for example in CLI context
+ */
+class LegacyContextLoader
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param Context $context
+     */
+    public function __construct(Context $context)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * @param string|null $controllerClassName
+     * @param int|null $currencyId
+     * @param int|null $employeeId
+     * @param int|null $shopId
+     * @param int|null $shopGroupId
+     *
+     * @return self
+     */
+    public function loadGenericContext(
+        ?string $controllerClassName = null,
+        ?int $currencyId = null,
+        ?int $employeeId = null,
+        ?int $shopId = null,
+        ?int $shopGroupId = null
+    ): self {
+        $this->loadCurrencyContext($currencyId);
+        $this->loadControllerContext($controllerClassName);
+        $this->loadEmployeeContext($employeeId);
+
+        if (null !== $shopId) {
+            $this->loadShopContext($shopId);
+        }
+        if (null !== $shopGroupId) {
+            $this->loadShopGroupId($shopGroupId);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param string|null $controllerClassName
+     *
+     * @return self
+     */
+    public function loadControllerContext(?string $controllerClassName = null): self
+    {
+        if (null === $controllerClassName) {
+            $this->context->controller = new DummyAdminController();
+
+            return $this;
+        }
+
+        if (!class_exists($controllerClassName)) {
+            throw new \RuntimeException(
+                sprintf(
+                    'Cannot load controller context for classname %s',
+                    $controllerClassName
+                )
+            );
+        }
+
+        $this->context->controller = new $controllerClassName();
+
+        return $this;
+    }
+
+    /**
+     * @param int|null $currencyId
+     *
+     * @return self
+     */
+    public function loadCurrencyContext(?int $currencyId = null): self
+    {
+        $currency = new Currency($currencyId);
+        if (null === $currencyId) {
+            $currency->precision = 2;
```

Comment:
```suggestion
            $currency->precision = Precision::DEFAULT_PRECISION;
```
Stored in `PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\ Precision`

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/21125/commits/98576208b2c085ca211f197109bb94781ef34679